vignettes/using_lintr.Rmd

```diff
@@ -0,0 +1,311 @@
+the argument `parse_settings = FALSE` can be added to the function calls. This will suppress reading of the `.lintr`
+configuration. This is particularly useful for tests which should not exclude example files containing lints while the
+package-level `.lintr` excludes those files because the lints are intentional.
+
+### Defaults
+
+The default settings of `lintr` are intended to conform to the [tidyverse style guide](https://style.tidyverse.org/).
+However, the behavior can be customized using different methods.
+
+Apart from `lintr.linter_file`, which defaults to `".lintr"`, there are the following settings:
+
+```{r, echo = FALSE}
+default_settings <- lintr::default_settings
+default_settings$linters <- "`lintr::default_linters`"
+default_settings$comment_token <- "(lintr-bot comment token for automatic GitHub comments)"
+default_settings$exclusions <- "(empty)"
+
+make_string <- function(x) {
+  if (inherits(x, "regex")) {
+    paste0("regex: `", x, "`")
+  } else {
+    as.character(x)
+  }
+}
+
+defaults_table <- data.frame(
+  default = vapply(default_settings, make_string, character(1L)),
+  stringsAsFactors = FALSE
+)
+
+knitr::kable(defaults_table)
+```
+
+#### Customizing active linters
+
+If one wants to only customize some linters, one can use the helper function `with_defaults()`, which will keep all
+untouched linters with the default settings. Disable a linter by passing `NULL`. For example, to set the line length
+limit to 120 characters and globally disable the `no_tab_linter`, you can put this into your `.lintr`:
+
+```r
+linters: with_defaults(
+    line_length_linter = line_length_linter(120L),
+    no_tab_linter = NULL
+  )
+```
+
+By default, the following linters are enabled. Where applicable, the default settings are also shown.
+
+```{r, echo = FALSE}
+default_linters <- lintr::default_linters
+linters_with_args <- list(
+  cyclocomp_linter = "complexity_limit",
+  line_length_linter = "length",
+  object_length_linter = "length",
+  closed_curly_linter = "allow_single_line",
+  object_name_linter = "styles",
+  open_curly_linter = "allow_single_line"
+)
+
+make_setting_string <- function(linter_name) {
+  args <- linters_with_args[[linter_name]]
+  if (is.null(args)) return("")
+
+  arglist <- vapply(args, function(arg) {
+    env <- environment(default_linters[[linter_name]])
+    if (linter_name == "object_length_linter") {
+      # object_length_linter creates a inner implementation named `fun`.
+      # all other linters keep their configuration in the same environment as the linter function.
+      env <- environment(env[["fun"]])
+    }
+    deparse(env[[arg]])
+  }, character(1L))
+
+  paste0(args, " = ", arglist, collapse = ", ")
+}
+
+defaults_table <- data.frame(
+  row.names = names(default_linters),
+  settings = vapply(names(default_linters), make_setting_string, character(1L)),
+  stringsAsFactors = FALSE
+)
+
+knitr::kable(defaults_table)
+```
+
+## Exclusions
```

Comment:
It might make sense to start narrowly (simple `# nolint`), then broaden to exclusion ranges (simple `# nolint start`), then broaden (in terms of code complexity) to specific exclusions (both `# nolint: <list>.` and `# nolint start: <list>.`). WDYT?

Reply:
I think we should actually promote narrow exclusions as best practice. The less you `# nolint` the higher the chance of catching quality issues...

Reply:
I meant more from the logical flow of the vignette -- it'll be easier to introduce one small concept at a time, no?

We can add a blurb to the top of this section mentioning the preferred approach, then build up to it.

Reply:
Okay, I'll try that.